src/extensions/changelog.ts

```diff
@@ -0,0 +1,85 @@
+import * as marked from "marked";
+import * as clc from "cli-color";
+import * as semver from "semver";
+import TerminalRenderer = require("marked-terminal");
+import Table = require("cli-table");
+
+import { listExtensionVersions, parseRef } from "./extensionsApi";
+import { logger } from "../logger";
+import { logLabeledWarning } from "../utils";
+
+marked.setOptions({
+  renderer: new TerminalRenderer(),
+});
+
+/*
+ * getReleaseNotesForUpdate fetches all version between toVersion and fromVersion and returns the relase notes
+ * for those versions if they exist.
+ * @param extensionRef
+ * @param fromVersion the version you are updating from
+ * @param toVersion the version you are upodating to
+ * @returns a Record of version number to releaseNotes for that version
+ */
+export async function getReleaseNotesForUpdate(
```

Comment:
prefer named params to reduce confusion at call site
go/tsstyle#comments-when-calling-a-function